src/ir_Bosch.h

```diff
@@ -0,0 +1,166 @@
+    uint8_t            :5;   // not used (without timer use)             # 
+    uint8_t FanS2      :3;   // Fan speed bits in Section 2              #  
+    uint8_t InnvertS2_2:8;   // Invert byte                              # Section 2 = Sektion 1
+    uint8_t            :2;   // not used (without timer use)             #
+    uint8_t ModeS2     :2;   // Operation mode bits S2                   #
+    uint8_t TempS2     :4;   // Desired temperature (Celsius) S2         #
+    uint8_t InnvertS2_3:8;   // Invert byte (without timer use)          ###########
+
+    uint8_t            :8;   //< Fixed value 0b11010101 / 0xD5           ###########
+    uint8_t ModeS3     :1;   // ModeBit in Section 3                     #
+    uint8_t FanS3      :6;   // Fan speed bits in Section 3              #
+    uint8_t            :1;   // Unknown                                  #
+    uint8_t            :7;   // Unknown                                  #
+    uint8_t Quiet      :1;   // Silent-Mode                              #    Section 3
+    uint8_t            :4;   // Unknown                                  #
+    uint8_t TempS3     :1;   // Desired temp. Bit in Section3            #
+    uint8_t            :3;   // Unknown                                  #
+    uint8_t            :8;   // Unknown                                  #
+    uint8_t ChecksumS3 :8;   // Checksum from byte 13-17                 ###########
+  };
+};
+
+// Classes
+
+/// Class for handling detailed Bosch144 A/C messages.
+class IRBosch144AC {
+ public:
+  explicit IRBosch144AC(const uint16_t pin, const bool inverted = false,
+                      const bool use_modulation = true);
+  void stateReset(void);
+#if SEND_BOSCH144
+  void send(const uint16_t repeat = 0);
+  /// Run the calibration to calculate uSec timing offsets for this platform.
+  /// @return The uSec timing offset needed per modulation of the IR Led.
+  /// @note This will produce a 65ms IR signal pulse at 38kHz.
+  ///   Only ever needs to be run once per object instantiation, if at all.
+  int8_t calibrate(void) { return _irsend.calibrate(); }
+#endif  // SEND_BOSCH144
+  void begin();
+  void setPower(const bool state);
+  bool getPower(void) const;
+  void setTemp(const uint8_t temp);
+  uint8_t getTemp(void) const;
+  void setFan(const uint16_t speed);
+  uint16_t getFan(void) const;
+  void setMode(const uint8_t mode);
+  uint8_t getMode(void) const;
+  void setQuiet(const bool on);
+  bool getQuiet(void) const;
+  uint8_t* getRaw(void);
+  void setRaw( const uint8_t new_code[] );
+  static uint8_t convertMode(const stdAc::opmode_t mode);
+  static uint16_t convertFan(const stdAc::fanspeed_t speed);
+  static stdAc::opmode_t toCommonMode(const uint8_t mode);
+  static stdAc::fanspeed_t toCommonFanSpeed(const uint16_t speed);
+  stdAc::state_t toCommon(void) const;
+  String toString(void) const;
+  void setInvertBytes();
+  void setCheckSumS3();
```

Comment:
Do these need to be public? Especially if they are added to `getRaw()`

Reply:
changed